src/main/java/seedu/address/model/person/position/Student.java

```diff
@@ -50,31 +61,58 @@ public static boolean isValidAttendance(String test) {
         }
     }
 
+    /**
+     * Returns true if a given string is a valid string of Assignments
+     */
+    public static boolean isValidAssignments(String test) {
+        String[] splitStr = test.split(", ");
+        int len = splitStr.length;
+        int totalWeightage = 0;
+
+        for (int i = 0; i < len; i++) {
+            String[] weightageStr = splitStr[i].split("w/");
+            if (weightageStr.length != 2) {
+                return false;
+            }
+            int weightage = Integer.parseInt(weightageStr[1]);
+            if (weightage < 0) {
+                return false;
+            }
+            totalWeightage += weightage;
+        }
+
+        return totalWeightage == 100;
+    }
+
     public void setGrade(String grade) {
         requireNonNull(grade);
         this.grade = grade;
     }
     @Override
     public void setDetails(String details) {
-        String[] gradeAndAttendance = isolateGradeAndAttendance(details);
+        String[] gradeAndAttendance = isolateDetails(details);
         String grade = gradeAndAttendance[0];
         String attendance = gradeAndAttendance[1];
+        String assignments = gradeAndAttendance[2];
         setGrade(grade);
         setAttendance(attendance);
+        setPreviousAssignments(assignments);
     }
 
     /**
      * method used to isolate the grade and attendance values in its details
```

Comment:
Can be rephrased to "Isolates the grade, attendance, and assignments values in the details"